libs/ui-components/src/components/Fleet/FleetsPage.tsx

```diff
@@ -34,19 +34,26 @@ import { ROUTE, useNavigate } from '../../hooks/useNavigate';
 import DeleteFleetModal from './DeleteFleetModal/DeleteFleetModal';
 import FleetResourceSyncs from './FleetResourceSyncs';
 import { useFleetBackendFilters, useFleets } from './useFleets';
+import { useAccessReview } from '../../hooks/useAccessReview';
+import { RESOURCE, VERB } from '../../hooks/useAppContext';
 
 const FleetPageActions = ({ createText }: { createText?: string }) => {
   const { t } = useTranslation();
   const navigate = useNavigate();
+  const [isCreateAllowed, isCreateLoading, createCheckError] = useAccessReview(RESOURCE.FLEET, VERB.CREATE);
+  const [isImportAllowed, isImportLoading, importCheckError] = useAccessReview(RESOURCE.REPOSITORY, VERB.CREATE);
+
+  const isCreateDisabled = !isCreateAllowed || isCreateLoading || !!createCheckError;
+  const isImportDisabled = !isImportAllowed || isImportLoading || !!importCheckError;
   return (
     <Split hasGutter>
       <SplitItem>
-        <Button variant="primary" onClick={() => navigate(ROUTE.FLEET_CREATE)}>
+        <Button variant="primary" onClick={() => navigate(ROUTE.FLEET_CREATE)} isDisabled={isCreateDisabled}>
```

Comment:
Nit: I'm wondering if it would be a good idea to create a component for the button + the access check.
We may want to include a message for the disabled buttons, and that would help DRY.

Reply:
FYI after discussion with Nir, we decided to hide the elements instead of disabling them.